awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx

```diff
@@ -1,14 +1,209 @@
+import { useLocation, useRouteMatch } from 'react-router-dom';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
 import { Card, PageSection } from '@patternfly/react-core';
 
-function InstanceGroupList() {
+import { InstanceGroupsAPI } from '../../../api';
+import { getQSConfig, parseQueryString } from '../../../util/qs';
+import useRequest, { useDeleteItems } from '../../../util/useRequest';
+import useSelected from '../../../util/useSelected';
+import PaginatedDataList, {
+  ToolbarDeleteButton,
+  ToolbarAddButton,
+} from '../../../components/PaginatedDataList';
+import ErrorDetail from '../../../components/ErrorDetail';
+import AlertModal from '../../../components/AlertModal';
+import DatalistToolbar from '../../../components/DataListToolbar';
+import InstanceGroupListItem from './InstanceGroupListItem';
+
+const QS_CONFIG = getQSConfig('instance_group', {
+  page: 1,
+  page_size: 20,
+});
+
+const modifyInstanceGroups = (items = []) => {
+  return items.map(function updateDelete(item) {
+    if (item.name === 'tower') {
+      item.summary_fields.user_capabilities.delete = false;
+    }
+    return item;
+  });
+};
+
+function InstanceGroupList({ i18n }) {
+  const location = useLocation();
+  const match = useRouteMatch();
+
+  const {
+    error: contentError,
+    isLoading,
+    request: fetchInstanceGroups,
+    result: { instanceGroups, instanceGroupsCount, actions },
+  } = useRequest(
+    useCallback(async () => {
+      const params = parseQueryString(QS_CONFIG, location.search);
+
+      const [response, responseActions] = await Promise.all([
+        InstanceGroupsAPI.read(params),
+        InstanceGroupsAPI.readOptions(),
+      ]);
+
+      return {
+        instanceGroups: response.data.results,
+        instanceGroupsCount: response.data.count,
+        actions: responseActions.data.actions,
+      };
+    }, [location]),
+    {
+      instanceGroups: [],
+      instanceGroupsCount: 0,
+      actions: {},
+    }
+  );
+
+  useEffect(() => {
+    fetchInstanceGroups();
+  }, [fetchInstanceGroups]);
+
+  const { selected, isAllSelected, handleSelect, setSelected } = useSelected(
+    instanceGroups
+  );
+
+  const modifiedSelected = modifyInstanceGroups(selected);
+
+  const {
+    isLoading: deleteLoading,
+    deletionError,
+    deleteItems: handleDeleteInstanceGroups,
+    clearDeletionError,
+  } = useDeleteItems(
+    useCallback(async () => {
+      await Promise.all(
+        selected.map(({ id }) => InstanceGroupsAPI.destroy(id))
+      );
+    }, [selected]),
+    {
+      qsConfig: QS_CONFIG,
+      allItemsSelected: isAllSelected,
+      fetchItems: fetchInstanceGroups,
+    }
+  );
+
+  const handleDelete = async () => {
+    await handleDeleteInstanceGroups();
+    setSelected([]);
+  };
+
+  const canAdd = actions && actions.POST;
+
+  function cannotDelete(item) {
+    return !item.summary_fields.user_capabilities.delete;
+  }
+
+  let itemsUnableToDelete = false;
```

Comment:
I don't see where this is being used in this component

Reply:
@AlexSCorey here. Are you mentioning `itemsUnableToDelete`? See line 115.

Reply:
Updated.